ansible/playbooks/roles/jmx_exporter/tasks/upgrade/main.yml

```diff
@@ -0,0 +1,101 @@
+---
+
+- name: Check for upgrade flag file
+  stat:
+    path: "{{ lock_file }}"
+  register: lock_file_status
+
+- name: Preflight check
+  when: not lock_file_status.stat.exists
+  block:
+  - name: Check for binary
+    stat:
+      path: "{{ prometheus_jmx_exporter_path }}"
+    register: jmx_exporter_binary
+
+  - name: Inform and fail if JMX exporter binary is missing
+    when: not jmx_exporter_binary.stat.exists
+    fail:
+      msg: >
+        There is no JMX exporter binary at all,
+        re-apply your configuration or remove jmx_exporter role from inventory
+
+  - name: Verify installed version
+    when: jmx_exporter_binary.stat.exists
+    block:
+    - name: Get installed jmx-exporter version
+      shell: >-
+        unzip -p /opt/jmx-exporter/jmx_prometheus_javaagent.jar META-INF/MANIFEST.MF \
+        | awk '$1 == "Implementation-Version:" {print $2}'
+      register: installed_jmx_exporter_version
+
+    - name: Set version facts
+      set_fact:
+        jmx_exporter_version:
+          old: "{{ installed_jmx_exporter_version.stdout }}"
+          new: "{{ jmx_exporter_version }}"
+
+    - name: Avoiding risk of downgrade
+      when: jmx_exporter_version.old is not version( jmx_exporter_version.new, '<' )
+      debug:
+        msg: "Skipping upgrade: JMX Exporter in newer version already installed!"
```

Comment:
@cicharka 
This can be a bit confusing when you have the same version.
After upgrading I have v0.16.1 and on re-run it tells me that `JMX Exporter in newer version is already installed!`.
Actually it's not newer, but the same.
Or the condition should be changed to `<=`.

Reply:
Fixed in #2794